ratelimits/limit.go

```diff
@@ -0,0 +1,91 @@
+package ratelimits
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/letsencrypt/boulder/config"
+	"github.com/letsencrypt/boulder/strictyaml"
+)
+
+type limit struct {
+	// Burst specifies maximum concurrent allowed requests at any given time. It
+	// must be greater than zero.
+	Burst int64
+
+	// Count is the number of requests allowed per period. It must be greater
+	// than zero.
+	Count int64
+
+	// Period is the duration of time in which the count (of requests) is
+	// allowed. It must be greater than zero.
+	Period config.Duration
+}
+
+type limits map[string]limit
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface.
+func (l *limits) UnmarshalYAML(unmarshal func(interface{}) error) error {
```

Comment:
With the strictyaml.UnmarshalYAML, this seems like it should instead be a validation method. Use `os.ReadFile` to get the config bytes, send those bytes to strictyaml, then perform validations over that.

Reply:
Phil and I went over this comment together offline.